src/compiler/checker.ts

```diff
@@ -38001,7 +38001,7 @@ namespace ts {
                     const childSymbol = getSymbolAtLocation(childNode);
                     if (childSymbol && childSymbol === testedSymbol) {
                         // If the test was a simple identifier, the above check is sufficient
-                        if (isIdentifier(expr)) {
+                        if (isIdentifier(expr) || isIdentifier(testedNode) && isBinaryExpression(testedNode.parent)) {

```

Comment:
What doesn't work if you just rewrite this as

```suggestion
                        if (isIdentifier(testedNode) && isBinaryExpression(testedNode.parent)) {
```

Reply:
`if (onChange) onChange()`